### ISSUE TITLE
Indoor low fps

### ISSUE THREAD

**Issue by bugproof** [Try/OpenGothic#280]
[opengothic-v1.0.1527](https://github.com/Try/OpenGothic/releases/tag/opengothic-v1.0.1527)

Indoor 40-60 fps
Outdoor 200-240 fps

**Comment by Try** [comment]
Hi, @bugproof !

Some detail about graphics hardware and game setup would be nice to have in description.
If, you want me to guess: I assume you have rayTracing-capable gpu, and since indoor areas have multiple lights, that can work as explanation

**Comment by bugproof** [comment]
I use RTX 2070S and windows 11 so raytracing is enabled by default?

**Comment by Try** [comment]
> raytracing is enabled by default?

Yes, for every discrete gpu that claim to support it.
you can add `-rt 0` to application command line - that will disable rayTracing

**Comment by bugproof** [comment]
Did `Gothic2Notr.exe -rt 0 -g "C:\Program Files (x86)\Steam\steamapps\common\Gothic II" >> log.txt` but didnt work.

**Comment by lmichaelis** [comment]
I have the same issue but I can confirm that it's todo with raytracing. The `-rt` flag is not implemented correctly:

https://github.com/Try/OpenGothic/blob/15d89bca24c04bfdf287d9b5ec12a65c9cb69a49/game/commandline.cpp#L71-L74

There is a `++i` missing which causes the `std::string_view(argv[i])!="0"` check to always compare `"-rt" != "0"` which is always `true`. After adding `++i` before the `isRQuery=` statement and passing `-rt 0` as a command-line parameter, performance is back to what it was before.

RT does not seem mature enough though. I am running an RX6800 XT and I get about 20 FPS indoors. Maybe it should be opt-in?

**Comment by sirsteve1** [comment]
I do can confirm the problem with the low fps. When I start the game I run around within Xardas tower with an fps of around 30 - this makes the game really ugly. 

- Ubuntu 22
- Nvidia GTX 920M
- Vulkan and Nvidia are working well

I tried to change practically all graphical configurations to a minimum without any change of fps (regardless of least or best configuration). I also added the statement of **lmichaelis** recompiling with the ++i statement in commandline.cpp to really deactivate rt, but this also does not seem to change anything. 

Any ideas?

I previously played Legend of Ahssun using wine which worked pretty well using the standard rendering engine.

**Comment by lmichaelis** [comment]
Hi @sirsteve1, you can try [this patch](https://github.com/Try/OpenGothic/files/9859643/Fix_-rt_flag.txt). It should work at least for the RT issue. To apply it, download the file and place it in the _OpenGothic_ repository directory on your disk. Open a terminal there and run `git apply Fix_-rt_flag.txt`. `git diff` will show the change made :) Don't forget to recompile and add `-rt 0` when running _OpenGothic_.

**Comment by sirsteve1** [comment]
> Hi @sirsteve1, you can try [this patch](https://github.com/Try/OpenGothic/files/9859643/Fix_-rt_flag.txt). It should work at least for the RT issue. To apply it, download the file and place it in the _OpenGothic_ repository directory on your disk. Open a terminal there and run `git apply Fix_-rt_flag.txt`. `git diff` will show the change made :) Don't forget to recompile and add `-rt 0` when running _OpenGothic_.

Thank you, but I've already done that myself and recompiled. As mentioned above this does not seem to fix the problem for me. Maybe its an overall nvidia-vulkan problem in itself. I will check out if there are other possible configurations and write back if I find something. Up until now the game seems to run only in 30 - 40 fps in generel for me.

**Comment by sirsteve1** [comment]
Managed to make the game work. The problem was not with the game itself nor raytracing for me. **The problem was the display driver**. Due to a problem with an update of the X-display driver I switched to wayland some weeks ago. When using wayland, the whole game is glitchy and flickering when moving around. I switched to the X-display session and started the game, everything is now working smoothly (though the fps is still low). But the game is playable and looks nice. I also activated all possible configurations except of cloud shadows - I even activated ray tracing - and the game still works smoothly. 

So its good to have mutliple display drivers to switch from time to time for best results. 
For me Wayland currently has problems with vulkan, but I read about this and guess its still in development with Nvidia.
